app/models/rules_of_origin/link.rb

```diff
@@ -1,7 +1,13 @@
 require 'api_entity'
+require 'digest'
```

Comment:
Don't think you need to require this?

Reply:
I'll try it without

Reply:
I'm going to leave it - from the look of it we're not requiring elsewhere in the site but Rails does require it so I _think_ thats why it still works. Relying on an implicit require within rails, that wont be autoloaded feels a bit fragile to me so I'll leave it here. It does no harm and makes the dependency explicit.